### ISSUE TITLE
create new node action does not work properly in context menu of jstree jquery plugin

### ISSUE THREAD

**Issue by ahmadbadpey** [vakata/jstree#1395]
I am using jstree jquery plugin to create a treeview list.

Now I want use context menu plugin for that and I want to change only label of menu items.

according to this Answer , I wrote this codes :

```
$(document).ready(function () {
    $treeview = $('#jstree_demo_div');

    $treeview.on("loaded.jstree", function (event, data) {
        $(this).jstree("open_all");
    });

    $treeview.jstree({
        "contextmenu": {
            "items": function ($node) {
                return {
                    "Create": {
                        "label": "create new Cat",
                        "action": function (data) {
                            var ref = $.jstree.reference(data.reference);
                            sel = ref.get_selected();
                            if(!sel.length) { return false; }
                            sel = sel[0];
                            sel = ref.create_node(sel, {"type":"file"});
                            if(sel) {
                                ref.edit(sel);
                            }
                        }
                    },
                    "Rename": {
                        "label": "rename ",
                        "action": function (data) {
                            var inst = $.jstree.reference(data.reference);
                            obj = inst.get_node(data.reference);
                            inst.edit(obj);
                        }
                    },
                    "Remove": {
                        "label": "delete",
                        "action": function (data) {
                            var ref = $.jstree.reference(data.reference),
                                sel = ref.get_selected();
                            if(!sel.length) { return false; }
                            ref.delete_node(sel);

                        }
                    }
                };
            }
        },
        "core": {
            "check_callback": true
        },
        "checkbox": {
            "keep_selected_style": false
        },
        "plugins": ["checkbox", "dnd" , "contextmenu"]
    });


});
```
```
<script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.1/jquery.min.js"></script>
<link href="https://cdnjs.cloudflare.com/ajax/libs/jstree/3.2.1/themes/default/style.min.css" rel="stylesheet"/>
<script src="https://cdnjs.cloudflare.com/ajax/libs/jstree/3.2.1/jstree.min.js"></script>
<div id="jstree_demo_div" style="direction: rtl">
        <ul>
            <li>sport
                <ul>
                    <li><a href="#">football</a></li>
                    <li><a href="#">basketbal</a></li>
                </ul>
            </li>
            <li>learning
                <ul>
                    <li><a href="#">web</a></li>
                    <li><a href="#">html</a></li>
                    <li><a href="#">css</a></li>
                </ul>
            </li>
        </ul>

    </div>
```

_remove_ and _rename_ defined actions work fine , but _Create_ have a problem.

First time, when I click on a node to create a sub node for it, all things works properly but when I select another node to create new sub node, the new node is created for previous selected node.

**Comment by vakata** [comment]
This is because you are always using the first selected node as a reference - from what I get - you have multiple selection enabled. Use the actual node that was right-clicked, not the first selected node. Basically - use `$node` and not `sel[0]`, if that does not work - please provide a jsfiddle demo.

Best regards,
Ivan
